File: ProblemSet4/EnumTest.swift
enum HospitalCodes: String
{
    case blue = "blue"
    case red = "red"
    case orange = "orange"
    case silver = "silver"
    case grey = "grey"
    case amberAlert = "amberAlert"
    case internalTriage = "internalTriage"
    case externalTriage = "externalTriage"
    case rapidResponseTeam = "rapidResponseTeam"
    case clear = "clear"
}

var emergencyCode: String;
emergencyCode = HospitalCodes.blue.rawValue;
print("Code: \(emergencyCode)");

emergencyCode = HospitalCodes.red.rawValue;
print("Code: \(emergencyCode)");

emergencyCode = HospitalCodes.orange.rawValue;
print("Code: \(emergencyCode)");

emergencyCode = HospitalCodes.silver.rawValue;
print("Code: \(emergencyCode)");

emergencyCode = HospitalCodes.grey.rawValue;
print("Code: \(emergencyCode)");

emergencyCode = HospitalCodes.amberAlert.rawValue;
print("Code: \(emergencyCode)");

emergencyCode = HospitalCodes.internalTriage.rawValue;
print("Code: \(emergencyCode)");

emergencyCode = HospitalCodes.externalTriage.rawValue;
print("Code: \(emergencyCode)");

emergencyCode = HospitalCodes.rapidResponseTeam.rawValue;
print("Code: \(emergencyCode)");

emergencyCode = HospitalCodes.clear.rawValue;
print("Code: \(emergencyCode)");

File: ProblemSet4/EnumTest.c
#include <stdio.h>
#include <stdlib.h>

#define EMERGENCY_TEST red

enum HospitalCodes
{
	blue,
	red,
	orange,
	silver,
	grey,
	amberAlert,
	internalTriage,
	externalTriage,
	rapidResponseTeam,
	clear
};

int main(void)
{
	enum HospitalCodes currentEmergencyCode = EMERGENCY_TEST;
	enum HospitalCodes clearEmergencyCode;

	if(currentEmergencyCode == red)
	{
		printf("Alert(%d): Fire!!!\n", currentEmergencyCode);
	}

	if(currentEmergencyCode == 1)
	{
		printf("Alert(%d): Fire!!!\n", currentEmergencyCode);
	}

	printf("\nSince the above statement printed twice, C enums are coerced to integers starting at index 0.\n");

	/*
	 * Nonsensical enum operation
	 */

	clearEmergencyCode = externalTriage + orange;

	if(clearEmergencyCode == clear)
	{
		printf("\nAlert(%d): All Clear\n", clearEmergencyCode);
	}

	return(EXIT_SUCCESS);
}